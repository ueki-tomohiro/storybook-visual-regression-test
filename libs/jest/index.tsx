import { composeStories, StoryFile } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import React from "react";

export const renderFromStories = (stories: StoryFile) => {
  const composed = composeStories(stories);
  const testCases = Object.values(composed).map((Story: any) => [Story.storyName, Story, Story.args]);
  test.each(testCases)("renders %s", (_storyName, Component: any, args) => {
    if (_storyName.match(/Fake/)) return;

    const tree = render(<Component {...(args || {})} />);
    expect(tree.baseElement).toMatchSnapshot();
  });
};
