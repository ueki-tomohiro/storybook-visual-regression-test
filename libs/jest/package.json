{
  "name": "@demo-libs/jest",
  "version": "0.0.1",
  "private": true,
  "description": "Demo Jest",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.21.4",
    "@babel/preset-env": "^7.21.4",
    "@babel/preset-typescript": "^7.21.4",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.5.1",
    "@types/jest": "^29.5.1",
    "babel-jest": "^29.5.0",
    "babel-plugin-transform-import-meta": "^2.2.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0"
  },
  "engines": {
    "node": "^20",
    "yarn": ">=1"
  },
  "volta": {
    "node": "20.0.0",
    "yarn": "1.22.19"
  }
}
