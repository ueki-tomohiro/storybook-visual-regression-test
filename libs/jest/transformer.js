const babelJest = require("babel-jest");
const babelOptions = {
  presets: [
    [
      "@babel/env",
      {
        targets: {
          node: "current",
        },
        exclude: ["transform-regenerator"],
      },
    ],
    "@babel/preset-typescript",
    "@babel/react",
  ],
  plugins: ["babel-plugin-transform-import-meta"],
  babelrc: false,
};
module.exports = babelJest.createTransformer(babelOptions);
