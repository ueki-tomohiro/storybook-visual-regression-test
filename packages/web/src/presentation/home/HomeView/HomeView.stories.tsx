import { getGetTodosMock } from "@demo/api/lib/demo/todo/todo.msw";

import { HomeView, HomeViewProps } from "./HomeView";

const defaultArgs: HomeViewProps = {
  todos: [
    {
      todo_id: 63460,
      description: "HTTP",
      end_date: "2022-12-07T12:13:58.000Z",
      completed: false,
    },
    {
      todo_id: 43073,
      description: "parse",
      end_date: "2023-01-21T22:30:27.000Z",
      completed: true,
    },
    {
      todo_id: 49534,
      description: "apud",
      end_date: "2022-07-06T18:01:16.000Z",
      completed: false,
    },
  ],
  isLoading: false,
};

export default {
  title: "views/HomeView",
  component: HomeView,
};

export const Default = { args: defaultArgs };

export const Loading = { args: { isLoading: true } };

export const Fake = {
  args: {
    ...defaultArgs,
    todos: getGetTodosMock(),
  },
};
