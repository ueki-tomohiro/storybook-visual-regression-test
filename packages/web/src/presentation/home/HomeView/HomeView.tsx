import { Todo } from "@demo/api/lib/demo/model";
import { LinkButton } from "@demo/ui/components/LinkButton";
import React from "react";

import { TodoItem } from "../TodoItem";

export type HomeViewProps = {
  todos?: Todo[];
  isLoading?: boolean;
};

export const HomeView: React.FC<HomeViewProps> = ({ isLoading, todos }) => {
  if (isLoading) {
    return <div>Loading...</div>;
  }
  return (
    <>
      <LinkButton href="/register" variant="primary">
        登録
      </LinkButton>
      <div className="mt-4">
        {todos?.map((todo) => (
          <TodoItem key={todo.todo_id} todo={todo} />
        ))}
      </div>
    </>
  );
};
