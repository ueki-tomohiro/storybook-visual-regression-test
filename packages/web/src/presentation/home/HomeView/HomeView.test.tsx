import { renderFromStories } from "@demo-libs/jest";

import * as stories from "./HomeView.stories";

describe(stories.default.title, () => {
  renderFromStories(stories);
});
