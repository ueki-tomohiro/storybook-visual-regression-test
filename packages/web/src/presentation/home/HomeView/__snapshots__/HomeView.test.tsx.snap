// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/HomeView renders Default 1`] = `
<body>
  <div>
    <a
      class="mx-auto max-w-3xl text-white bg-primary rounded px-4 py-2"
      href="/register"
    >
      登録
    </a>
    <div
      class="mt-4"
    >
      <a
        class="w-full"
        href="/todo/63460"
      >
        <div
          class="my-2 flex items-center border-2 p-2"
        >
          <div>
            <span
              class="material-symbols-outlined"
            >
              label_important
            </span>
          </div>
          <div
            class="mx-2 text-xl"
          >
            HTTP
          </div>
          <div
            class="text-xs text-tertiary"
          >
            2022-12-07 12:13:58
          </div>
        </div>
      </a>
      <a
        class="w-full"
        href="/todo/43073"
      >
        <div
          class="my-2 flex items-center border-2 p-2"
        >
          <div>
            <span
              class="material-symbols-outlined"
            >
              done
            </span>
          </div>
          <div
            class="mx-2 text-xl"
          >
            parse
          </div>
          <div
            class="text-xs text-tertiary"
          >
            2023-01-21 22:30:27
          </div>
        </div>
      </a>
      <a
        class="w-full"
        href="/todo/49534"
      >
        <div
          class="my-2 flex items-center border-2 p-2"
        >
          <div>
            <span
              class="material-symbols-outlined"
            >
              label_important
            </span>
          </div>
          <div
            class="mx-2 text-xl"
          >
            apud
          </div>
          <div
            class="text-xs text-tertiary"
          >
            2022-07-06 18:01:16
          </div>
        </div>
      </a>
    </div>
  </div>
</body>
`;

exports[`views/HomeView renders Loading 1`] = `
<body>
  <div>
    <div>
      Loading...
    </div>
  </div>
</body>
`;
